main.js

```diff
@@ -2920,10 +2920,10 @@ function startInstance(id, wakeUp) {
     }
 
     // if less than 100 MB log warning, less than 50 error
-    if (availableMemMB < 50) {
+    if (availableMemMB < 50 && availableMemMB !== undefined) {
```

Comment:
Why in this order?
```js
if (availableMemMB !== undefined && availableMemMB < 50)
```
is more logical IMO

Reply:
Because as we already have evaluated the second evaluation is unnecessary and will then only be evaluated if available mem is really under 50 or undefined, which will be the edge case. If we would really depend on undefined, then the order should be the other way round

Reply:
I agree its unnecessary but its more logical. I prefer readability and logic over performance arguments, especially if the code is executed once and we only do a single extraneous comparison.

P.S.: On Windows systems, `availableMemMB !== undefined` is always false, so we would always avoid doing the implicit comparison.

Reply:
Windows will hold the value of `os.freemem()` in MB  in the variable and thus won't be undefined. As said before, normally I would agree. However, IMHO, in this case, it doesn't make a difference in readability. We can change if you find it necessary ..